great_expectations/metrics/registry.py

```diff
@@ -0,0 +1,11 @@
+from typing import Final
+
+from great_expectations.metrics.domain import ColumnMap, Domain
+
+DOMAIN_NAMES: Final[dict[type[Domain], str]] = {
+    ColumnMap: "column_values",
+}
+
+METRIC_REGISTRY: Final[dict[str, set[str]]] = {
+    DOMAIN_NAMES[ColumnMap]: {"between"},
+}
```

Comment:
what value do we get out of having a global registry? I see two uses here:
- a place to store the name of a domain class
- check that contributors are correctly wiring up metrics to domains

For the first case, i think we're better served by just having a `name` attribute on the domain class. I'm not convinced the second use case is actually adding any value, since a) the check is aimed at library authors, not users, and b) the check can be easily circumvented by just adding a magic string to this file.

It's possible that i'm missing a path that makes this pattern valuable, so help me fill in the gaps if that's the case!

Reply:
I went ahead and nuked the registry. It will matter how we name classes now, but the existing metrics system will complain if that is done incorrectly.

Reply:
If we don't like the existing names as class names we can add a `name` override string